python/pyspark/sql/dataframe.py

```diff
@@ -347,13 +347,26 @@ def show(self, n=20, truncate=True, vertical=False):
          name | Bob
         """
         if isinstance(truncate, bool) and truncate:
-            print(self._jdf.showString(n, 20, vertical))
+            print(self._jdf.showString(n, 20, vertical, False))
         else:
-            print(self._jdf.showString(n, int(truncate), vertical))
+            print(self._jdf.showString(n, int(truncate), vertical, False))
 
     def __repr__(self):
         return "DataFrame[%s]" % (", ".join("%s: %s" % c for c in self.dtypes))
 
+    def _repr_html_(self):
```

Comment:
Add comments above this to explain this goal of this function.

Reply:
Thanks, done. feb5f4a.